typescript/token/contracts/FastHypERC20.sol

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0;
+
+import {TokenRouter} from "./libs/TokenRouter.sol";
+import {Message} from "./libs//Message.sol";
+
+import {ERC20Upgradeable} from "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+
+/**
+ * @title Hyperlane ERC20 Token Router that extends ERC20 with remote transfer functionality.
+ * @author Abacus Works
+ * @dev Supply on each chain is not constant but the aggregate supply across all chains is.
+ */
+contract FastHypERC20 is ERC20Upgradeable, TokenRouter {
+    uint8 private immutable _decimals;
+    uint256 public fastTransferId;
```

Comment:
This should work both ways

Reply:
Ohh so on the remote chain someone can provide the HypERC20 as liquidity?

Reply:
Yes